src/H5FDmpio.c

```diff
@@ -3097,9 +3097,9 @@ H5FD__mpio_read_selection(H5FD_t *_file, H5FD_mem_t type, hid_t H5_ATTR_UNUSED d
              * s_bufs[] to find the smallest value, and choose that for
              * mpi_bufs_base.
              */
-            j = 0; /* guess at the index of the smallest value of s_bufs[] */
 
-            if (s_bufs[j + 1].vp != NULL) {
+            j = 0; /* guess at the index of the smallest value of s_bufs[] */
+            if ((count > 1) && (s_bufs[1].vp != NULL)) {
```

Comment:
Have you tried removing the if statement entirely?

Reply:
Yes, removing the if statement lead to warnings about use of uninitialized entries in `s_bufs` due to some optimization that is being applied in `H5FD_sort_selection_io_req` that appears to skip setting entries beyond `s_bufs[0]` and `s_bufs[1]` to anything.